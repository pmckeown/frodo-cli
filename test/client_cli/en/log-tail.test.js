import cp from 'child_process';
import { promisify } from 'util';

const exec = promisify(cp.exec);

test("CLI help interface for 'log tail' should be expected english", async () => {
  const CMD = 'frodo log tail --help';
  const { stdout } = await exec(CMD);
  expect(stdout).toMatchSnapshot();
});

test("CLI help interface for 'logs tail' should be expected english", async () => {
  const CMD = 'frodo logs tail --help';
  const { stdout } = await exec(CMD);
  expect(stdout).toMatchSnapshot();
});
