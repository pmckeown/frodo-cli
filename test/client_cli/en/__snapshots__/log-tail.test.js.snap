// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CLI help interface for 'log tail' should be expected english 1`] = `
"Usage: frodo log tail [options] [host] [username] [password]

Tail Identity Cloud logs.

Arguments:
  host                         Access Management base URL, e.g.:
                               https://cdk.iam.example.com/am. To use a
                               connection profile, just specify a unique
                               substring.
  username                     Username to login with. Must be an admin user
                               with appropriate rights to manage authentication
                               journeys/trees.
  password                     Password.

Options:
  -c, --sources <sources>      Comma separated list of log sources (default:
                               Log everything)
  --curlirize                  Output all network calls in curl format.
  -d, --defaults               Use default logging noise filters (default: Use
                               custom logging noise filters defined in
                               $HOME/LoggingNoiseFilter.json)
  --debug                      Debug output during command execution. If
                               specified, may or may not produce additional
                               output helpful for troubleshooting.
  -h, --help                   Help
  -k, --insecure               Allow insecure connections when using SSL/TLS.
                               Has no effect when using a network proxy for
                               https (HTTPS_PROXY=http://<host>:<port>), in
                               that case the proxy must provide this
                               capability. (default: Don't allow insecure
                               connections)
  -l, --level <level>          Set log level filter. You can specify the level
                               as a number or a string. Following values are
                               possible (values on the same line are
                               equivalent):
                               0, SEVERE, FATAL, or ERROR
                               1, WARNING, WARN or CONFIG
                               2, INFO or INFORMATION
                               3, DEBUG, FINE, FINER or FINEST
                               4 or ALL (default: SEVERE,ERROR,FATAL)
  --sa-id <sa-id>              Service account id.
  --sa-jwk-file <file>         File containing the JSON Web Key (JWK)
                               associated with the the service account.
  -t, --transaction-id <txid>  Filter by transactionId
  --verbose                    Verbose output during command execution. If
                               specified, may or may not produce additional
                               output.

Evironment Variables:
  FRODO_HOST: Access Management base URL. Overrides 'host' argument.
  FRODO_REALM: Realm. Overrides 'realm' argument.
  FRODO_USERNAME: Username. Overrides 'username' argument.
  FRODO_PASSWORD: Password. Overrides 'password' argument.
  FRODO_SA_ID: Service account uuid. Overrides '--sa-id' option.
  FRODO_SA_JWK: Service account JWK. Overrides '--sa-jwk-file' option but takes the actual JWK as a value, not a file name.
  FRODO_AUTHENTICATION_SERVICE: Name of a login journey to use.
  FRODO_DEBUG: Set to any value to enable debug output. Same as '--debug'.

"
`;

exports[`CLI help interface for 'logs tail' should be expected english 1`] = `
"Usage: frodo log tail [options] [host] [username] [password]

Tail Identity Cloud logs.

Arguments:
  host                         Access Management base URL, e.g.:
                               https://cdk.iam.example.com/am. To use a
                               connection profile, just specify a unique
                               substring.
  username                     Username to login with. Must be an admin user
                               with appropriate rights to manage authentication
                               journeys/trees.
  password                     Password.

Options:
  -c, --sources <sources>      Comma separated list of log sources (default:
                               Log everything)
  --curlirize                  Output all network calls in curl format.
  -d, --defaults               Use default logging noise filters (default: Use
                               custom logging noise filters defined in
                               $HOME/LoggingNoiseFilter.json)
  --debug                      Debug output during command execution. If
                               specified, may or may not produce additional
                               output helpful for troubleshooting.
  -h, --help                   Help
  -k, --insecure               Allow insecure connections when using SSL/TLS.
                               Has no effect when using a network proxy for
                               https (HTTPS_PROXY=http://<host>:<port>), in
                               that case the proxy must provide this
                               capability. (default: Don't allow insecure
                               connections)
  -l, --level <level>          Set log level filter. You can specify the level
                               as a number or a string. Following values are
                               possible (values on the same line are
                               equivalent):
                               0, SEVERE, FATAL, or ERROR
                               1, WARNING, WARN or CONFIG
                               2, INFO or INFORMATION
                               3, DEBUG, FINE, FINER or FINEST
                               4 or ALL (default: SEVERE,ERROR,FATAL)
  --sa-id <sa-id>              Service account id.
  --sa-jwk-file <file>         File containing the JSON Web Key (JWK)
                               associated with the the service account.
  -t, --transaction-id <txid>  Filter by transactionId
  --verbose                    Verbose output during command execution. If
                               specified, may or may not produce additional
                               output.

Evironment Variables:
  FRODO_HOST: Access Management base URL. Overrides 'host' argument.
  FRODO_REALM: Realm. Overrides 'realm' argument.
  FRODO_USERNAME: Username. Overrides 'username' argument.
  FRODO_PASSWORD: Password. Overrides 'password' argument.
  FRODO_SA_ID: Service account uuid. Overrides '--sa-id' option.
  FRODO_SA_JWK: Service account JWK. Overrides '--sa-jwk-file' option but takes the actual JWK as a value, not a file name.
  FRODO_AUTHENTICATION_SERVICE: Name of a login journey to use.
  FRODO_DEBUG: Set to any value to enable debug output. Same as '--debug'.

"
`;
